app/controllers/listings_controller.rb

```diff
@@ -110,15 +110,19 @@ def show
 
     payment_gateway = MarketplaceService::Community::Query.payment_type(@current_community.id)
 
-    payment_process = select_payment_process(
-      price_field: @listing.transaction_type.price_field?,
-      preauthorize: @listing.transaction_type.preauthorize_payment?,
-      payment_gateway_available: payment_gateway.present?)
+    # TODO Change this so that the path is always the same, but the controller
+    # decides what to do. We don't want to make a API call to TransactionService
+    # just to show a listing details
+    process_res = TransactionService::API::Api.processes.get(
+      community_id: @current_community.id,
+      process_id: @listing.transaction_type.transaction_process_id
```

Comment:
Listing must record transaction_process_id reference at create/edit and use that. Is the plan to change that part at later? If so then perhaps add a TODO comment to not miss it later.